src/dash-table/handlers/cellEvents.ts

```diff
@@ -1,57 +1,69 @@
-import * as R from 'ramda';
-import { SelectedCells, ICellFactoryProps } from 'dash-table/components/Table/props';
+import { min, max, set, lensPath } from 'ramda';
+import { ICellFactoryProps } from 'dash-table/components/Table/props';
 import isActive from 'dash-table/derived/cell/isActive';
+import isSelected from 'dash-table/derived/cell/isSelected';
+import { makeCell, makeSelection } from 'dash-table/derived/cell/cellProps';
 import reconcile from 'dash-table/type/reconcile';
 
-function isCellSelected(selectedCells: SelectedCells, idx: number, i: number) {
-    return selectedCells && R.contains([idx, i], selectedCells);
-}
-
 export const handleClick = (propsFn: () => ICellFactoryProps, idx: number, i: number, e: any) => {
     const {
         selected_cells,
+        start_cell,
         setProps,
-        virtualized
+        virtualized,
+        columns,
+        viewport
     } = propsFn();
 
-    const selected = isCellSelected(selected_cells, idx, i);
+    const selected = isSelected(selected_cells, idx, i);
 
     // don't update if already selected
     if (selected) {
         return;
     }
```

Comment:
BTW in case of `shift-click` I think this return is problematic - it means you can't shrink a selection without doing something hacky like first clicking over to the other side of the starting point.

Reply:
better shift-click behavior -> https://github.com/plotly/dash-table/pull/412/commits/1c69cfbe8080524946f53d720ae5535886666136 (and https://github.com/plotly/dash-table/pull/412/commits/3e31a7f33418bc0f2b1830d6cf75f6e67830c736)